docs/Formula-Cookbook.md

```diff
@@ -881,9 +881,21 @@ There are two ways to add `launchd` plists and `systemd` services to a formula,
 2. If the formula does not provide a service file you can generate one using the following stanza:
 
    ```ruby
+   # 1. An individual command
    service do
      run opt_bin/"script"
    end
+
+   # 2. A command with arguments
+   service do
+     run [opt_bin/"script", "--config", etc/"dir/config.yml"]
+   end
+
+   # 3. OS specific commands (If you omit one, the service file won't get generated for that OS.)
+   service do
+     run macos: [opt_bin/"macos_script", "standalone"],
+         linux: var/"special_linux_script"
```

Comment:
This syntax seems to only be used for two formulae. Feel's overkill to even have this syntax instead of just wrapping the `service` in an `on_*` block given how little this is used.

Reply:
It looks like that was the first thing they tried but the linter didn't like it (https://github.com/Homebrew/brew/issues/14352). Now that the service block is included in the API this is needed to allow us to easily serialize platform specific run commands.

But yeah, I agree with you that there are some edge cases having to do with the service block that require extra complexity that maybe outweighs the number of formulae that are actually affected. #15396 is another example of this.

Reply:
That seems to be easily fixable by adjusting the cop.

Reply:
@reitermarkus I assume you're fine with me merging this in as is. The alternative would mean rethinking this OS specific service block API and rewriting the service block serialization code to accommodate that change.